Documentation about LavinMQ in-place shared to dedicated (#490)
cloudamqp/terraform-provider-cloudamqp: docs/guides/info_lavinmq_shared_to_dedicated.md

---
layout: "cloudamqp"
page_title: "CloudAMQP: LavinMQ shared to dedicated upgrade"
subcategory: "info"
description: |-
  Guide on how to upgrade a LavinMQ shared instance to a dedicated plan in-place.
---

# LavinMQ shared to dedicated upgrade

From v1.45.0 it is possible to upgrade a [**LavinMQ**] shared instance to a dedicated plan
in-place. The instance ID is preserved, no resource replacement occurs, and existing definitions
are kept. The upgrade is performed by changing the `plan` argument in your configuration and
running `terraform apply`.

~> This upgrade path is only available for **LavinMQ** instances. RabbitMQ shared instances cannot
be upgraded in-place to dedicated plans and will require a new resource.

## Shared LavinMQ plans

The following LavinMQ shared plans are eligible for in-place upgrade to a dedicated plan:

Plan       | Name
-----------|----------------
`lemming`  | Loyal Lemming
`ermine`   | Elegant Ermine

## Constraints

> * The cloud provider must remain the same. For example, moving between AWS regions is supported,
>   but moving from AWS to GCP is not.

Reviewer: ```suggestion
```
It is possible :) Or should be, maybe it isn't wiht the changes I made? 🙈

Reviewer: Okay, My changes did not accept this. BUt I think we can keep like this.

> * The target plan must be a dedicated LavinMQ plan. See available [plans].

Reviewer: And only single node available, to go to mulitnode. first upgrade to a single node :)

* The reverse upgrade path (dedicated to shared) is **not** supported and will force a new
  resource to be created.

## What can change during the upgrade

The following attributes can optionally be changed during the same `terraform apply` as the plan
change:

* `plan` — (Required) Must change to a dedicated LavinMQ plan (e.g. `puffin-1`, `penguin-1`,
  `wolverine-1`).

> * `region` — (Optional) Can change to a different region within the same cloud provider.

Reviewer: ```suggestion
* `region` — (Optional) Can change to a different region and cloud provider.
```
